aws/table_aws_ssm_document.go

```diff
@@ -359,6 +373,62 @@ func getAwsSSMDocument(ctx context.Context, d *plugin.QueryData, h *plugin.Hydra
 	return data.Document, nil
 }
 
+func getAwsSSMDocumentContent(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	var arn string
+	item, err := getAwsSSMDocumentArn(ctx, d, h)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_ssm_document.getAwsSSMDocumentContent", "arn_formatting_error", err)
+		return nil, err
+	}
+	arn = item.(string)
+
+	matrixRegion := d.EqualsQualString(matrixKeyRegion)
+	arnSplit := strings.Split(arn, ":") // Split ARN to get the region
+
+	// Invalid ARN check
+	if len(arnSplit) < 3 {
+		return nil, nil
+	}
+
+	// Skip ARNs in other regions
+	if matrixRegion != arnSplit[3] {
+		return nil, nil
+	}
+
+	name := strings.Split(arn, "/")[1] // Split ARN to get the document name
+
+	// Create Session
+	svc, err := SSMClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_ssm_document.getAwsSSMDocumentContent", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region check
+		return nil, nil
+	}
+
+	// Empty name input check
+	if strings.TrimSpace(name) == "" {
+		return nil, nil
+	}
+
+	// Build the params
+	params := &ssm.GetDocumentInput{
```

Comment:
Is there a way of passing the optional arguments for the `GetDocumentInput`? I see that we have other parameters like `DocumentVersion` and `DocumentFormat` in https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_GetDocument.html.

Could we consider the possibility of adding this as a separate table? From an API perspective, what happens when we pass in a different document version?

Reply:
Currently, it is not possible to add optional qualifiers for a hydrate call, even though the GetDocumentInput supports optional parameters.

> Could we consider the possibility of adding this as a separate table? From an API perspective, what happens when we pass in a different document version?

-  If we will add a separate table for it, we will always have a single row for a particular document version.
- The Document name  is required for making this API call, but DocumentVersion and DocumentFormat are optional.
- First we need a table that provides all available document versions for a specific document.
- As suggested in the issue comment, we have added this column in JSON format.